pkg/pulumiyaml/analyser.go

```diff
@@ -229,6 +229,10 @@ func (tc *typeCache) isAssignable(fromExpr ast.Expr, to schema.Type) *notAssigna
 		return nil
 	}
 
+	if from == schema.AnyType || to == schema.AnyType {
```

Comment:
Allowing an implicit cast from `any` to `T` is unsound. I would have liked to have more discussion about introducing a checked cast instead of just weakening the type checker.